ios/RNTrackPlayer/Models/MediaURL.swift

```diff
@@ -17,32 +17,21 @@ struct MediaURL {
         guard let object = object else { return nil }
         originalObject = object
         
+        // This is based on logic found in RCTConvert NSURLRequest, 
+        // and uses RCTConvert NSURL to create a valid URL from various formats
         if let localObject = object as? [String: Any] {
-            let uri = localObject["uri"] as! String
-            isLocal = uri.contains("http") ? false : true
-            let encodedURI = uri.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)!
-            value = URL(string: encodedURI.replacingOccurrences(of: "file://", with: ""))!
+            var url = localObject["uri"] as? String ?? localObject["url"] as! String
+            
+            if let bundleName = localObject["bundle"] as? String {
+                url = String(format: "%@.bundle/%@", bundleName, url)
+            }
+            
+            isLocal = url.lowercased().contains("http") ? false : true
```

Comment:
I think this would be safer with `.hasPrefix`, as the file path could somehow contain http, such as a random generated filename like this one: `MVHZSTGYNIHTTPSYUJBWVX.mp3`

Reply:
I agree, I will change this.

@dcvz added this line, so I was just trying to leave it untouched, but I don't see any notes or comments about why it would be the way it is instead of checking for prefix.